app/src/main/java/fr/free/nrw/commons/contributions/MainActivity.java

```diff
@@ -267,15 +253,8 @@ public void onBackPressed() {
             }
         } else if (getSupportFragmentManager().findFragmentByTag(nearbyFragmentTag) != null && !isContributionsFragmentVisible) {
             // Means that nearby fragment is visible (not contributions fragment)
-            NearbyFragment nearbyFragment = (NearbyFragment) contributionsActivityPagerAdapter.getItem(1);
-
-            if(nearbyFragment.isBottomSheetExpanded()) {
-                // Back should first hide the bottom sheet if it is expanded
-                nearbyFragment.listOptionMenuItemClicked();
-            } else {
-                // Otherwise go back to contributions fragment
-                viewPager.setCurrentItem(0);
-            }
+            NearbyParentFragment nearbyFragment = (NearbyParentFragment) contributionsActivityPagerAdapter.getItem(1);
```

Comment:
Again, the back press should be passed to the fragment and not the presenter directly

Reply:
As far as I understand from MVP documentations, user actions should be handled in presenter. I considered back button click as a user action, that's why I added it to presenter. Additionally, it controls some sorted operations.

Reply:
@neslihanturan Yes, the presenter should handle user actions but getting an instance of the fragment to get an instance of the presenter is not the right way of doing it. 

Maintain a singleton instance of the presenter which can be accessed without depending on fragment lifecycle.

Reply:
@neslihanturan Can you address this comment.

Reply:
Actually I have already addressed previous comment, please see changes at lines 271. As you suggested earlier, I used presenter as singleton. I just forgot to remove fragment instance (which is not in use anymore). Removing it now.